src/julia_gc.c

```diff
@@ -728,20 +728,36 @@ void InitBags(UInt initial_size, Bag * stack_bottom, UInt stack_align)
 
     // Import GAPTypes module to have access to GapObj abstract type.
     // Needs to be done before setting any GC states
-    jl_eval_string("import GAPTypes");
-    if (jl_exception_occurred()) {
-        Panic("could not import GAPTypes module into Julia");
+    UInt            go_to_gaptypes_fallback = 0;
+    jl_module_t *   gaptypes_module;
+    jl_datatype_t * gapobj_type = (jl_datatype_t *)jl_get_global(
+        jl_main_module, jl_symbol("__GapObj_TYPE_VAR"));
+    if (!jl_exception_occurred()) {
+        go_to_gaptypes_fallback = 1;
     }
-    // Get GapObj abstract julia type
-    jl_module_t * gaptypes_module =
-        (jl_module_t *)jl_get_global(jl_main_module, jl_symbol("GAPTypes"));
-    if (jl_exception_occurred()) {
-        Panic("Could not read global GAPTypes variable from Julia");
+    else {
+        gaptypes_module = (jl_module_t *)jl_get_global(
+            jl_main_module, jl_symbol("__GapObj_MODULE_VAR"));
+        if (jl_exception_occurred()) {
+            go_to_gaptypes_fallback = 1;
```

Comment:
Have you tested this? is there really an exception here. From the quick glance at the Julia code, my impression was that  `jl_get_global` returns NULL if the binding does not exist. Also, `jl_exception_occurred` w/o an explicit exception handler only works for a very, very small set of APIs explicitly designated for this purpose, and I am pretty sure `jl_get_global` is not one of them